firmware/common/pocsag.cpp

```diff
@@ -377,78 +380,80 @@ inline int errorCorrection(uint32_t* val) {
     return errl;
 }
 
-void pocsag_decode_batch(const POCSAGPacket& batch, POCSAGState* const state) {
-    int errors = 0;
-    uint32_t codeword;
-    char ascii_char;
-    std::string output_text = "";
-
-    state->out_type = EMPTY;
-
-    // For each codeword...
-    for (size_t i = 0; i < 16; i++) {
-        codeword = batch[i];
-
-        errorCorrection(&codeword);
-        errors = errorCorrection(&codeword);
-
-        if (!(codeword & 0x80000000U)) {
-            // Address codeword
-            if (state->mode == STATE_CLEAR) {
-                // if (codeword != POCSAG_IDLEWORD) {
-                if (!(bitsDiff(codeword, POCSAG_IDLEWORD) < 1)) {
-                    state->function = (codeword >> 11) & 3;
-                    state->address = (codeword >> 10) & 0x1FFFF8U;  // 18 MSBs are transmitted
-                    state->mode = STATE_HAVE_ADDRESS;
-                    state->out_type = ADDRESS;
-                    state->errors = errors;
-
-                    state->ascii_idx = 0;
-                    state->ascii_data = 0;
+bool pocsag_decode_batch(const POCSAGPacket& batch, POCSAGState& state) {
+    constexpr uint8_t codeword_max = 16;
+    state.output.clear();
+
+    while (state.codeword_index < codeword_max) {
+        auto codeword = batch[state.codeword_index];
+        bool is_message = (codeword & 0x80000000U) > 1;
```

Comment:
To just check whether this bit is set, I would suggest eliminating the "> 1", which I found confusing and would never be true if codeword happened to be a signed int. Perhaps:
`    bool is_message = (codeword & 0x80000000U) != 0;`

Reply:
I ended up reversing the logic here and it reads a lot better because all the tests used to be `!is_message` and are now `is_address`.